geode-core/src/main/java/org/apache/geode/internal/InternalDataSerializer.java

```diff
@@ -2684,9 +2605,9 @@ public static int readArrayLength(DataInput in) throws IOException {
       int result = ubyteToInt(code);
       if (result > MAX_BYTE_ARRAY_LEN) {
         if (code == SHORT_ARRAY_LEN) {
-          result = in.readUnsignedShort();
+          return in.readUnsignedShort();
```

Comment:
I don't like this change. It is confusing to have something called "result" but you end up returning something else. I'd go back to the old code or change the code to something like this:
int lengthFromCode = ubyteToInt(code);
if (lengthFromCode <= MAX_BYTE_ARRAY_LEN) {
  return lengthFromCode;
}
if (code == SHORT_ARRAY_LEN) {
  return in.readUnsignedShort();
...

Reply:
@dschneider-pivotal thank you, a much nicer change. I'll amend accordingly